File: app/src/main/java/com/makingiants/todayhistory/utils/NetworkChecker.java
package com.makingiants.todayhistory.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import timber.log.Timber;

public class NetworkChecker {

  private final ConnectivityManager mConnectivityManager;

  public NetworkChecker(Context context) {
    mConnectivityManager =
        (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
  }

  /**
   * https://gist.github.com/A7maDev/427694dcae675435ce53
   */
  public boolean isNetworkConnectionAvailable() {
    try {
      NetworkInfo networkInfo = mConnectivityManager.getActiveNetworkInfo();
      return networkInfo != null && networkInfo.isConnected() && networkInfo.isAvailable();
    } catch (Exception e) {
      Timber.e(e, "Getting internet connection status");
    }

    return false;
  }
}

File: app/src/main/kotlin/com/makingiants/todayhistory/utils/NetworkChecker.kt
package com.makingiants.todayhistory.utils

import android.content.Context
import android.net.ConnectivityManager
import timber.log.Timber

open class NetworkChecker(context: Context) {
    private val mConnectivityManager: ConnectivityManager

    init {
        mConnectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
    }

    /**
     * https://gist.github.com/A7maDev/427694dcae675435ce53
     */
    open fun isNetworkConnectionAvailable(): Boolean {
        try {
            val networkInfo = mConnectivityManager.activeNetworkInfo
            return networkInfo != null && networkInfo.isConnected && networkInfo.isAvailable
        } catch (e: Exception) {
            Timber.e(e, "Getting internet connection status")
        }

        return false
    }
}
